In [1]:
from nltk.corpus import stopwords 
from nltk.stem.wordnet import WordNetLemmatizer
import string
import pandas as pd

import gensim
from gensim import corpora

In [2]:
data = pd.read_csv("../Data/data.csv")

In [3]:
data['line'] = data['line'].str.lower()
documents = data['line'].tolist()

In [4]:
stop = set(stopwords.words('english'))
exclude = set(string.punctuation) 
lemma = WordNetLemmatizer()

In [5]:
def clean(doc):
    stop_free = " ".join([i for i in doc.lower().split() if i not in stop])
    punc_free = " ".join(ch for ch in stop_free if ch not in exclude)
    normalized = " ".join(lemma.lemmatize(word) for word in punc_free.split())
    return normalized

In [6]:
doc_clean = [clean(doc).split() for doc in documents]   

In [7]:
# term dictionary of courpus
dictionary = corpora.Dictionary(doc_clean)

# convert corpus into Document Term Matrix
doc_term_matrix = [dictionary.doc2bow(doc) for doc in doc_clean]

In [8]:
LDA = gensim.models.ldamodel.LdaModel

In [9]:
ldamodel = LDA(doc_term_matrix, num_topics=5, id2word = dictionary, passes=50)

In [12]:
print(ldamodel.print_topics(num_topics=5, num_words=5))

[(0, '0.037*"it" + 0.036*"know" + 0.016*"you" + 0.015*"im" + 0.015*"him"'), (1, '0.019*"get" + 0.016*"back" + 0.016*"right" + 0.016*"one" + 0.015*"it"'), (2, '0.021*"dead" + 0.019*"here" + 0.016*"dragon" + 0.015*"saw" + 0.013*"house"'), (3, '0.040*"you" + 0.029*"lord" + 0.027*"king" + 0.023*"queen" + 0.023*"come"'), (4, '0.052*"me" + 0.021*"want" + 0.019*"no" + 0.019*"dont" + 0.013*"see"')]


'0.037*"it" 
0.036*"know" 
0.016*"you" 
0.015*"im" 
0.015*"him"'

'0.019*"get" 
0.016*"back" 
0.016*"right" 
0.016*"one" 
0.015*"it"'

'0.021*"dead" 
0.019*"here" 
0.016*"dragon" 
0.015*"saw" 
0.013*"house"'

'0.040*"you" 
0.029*"lord" 
0.027*"king" 
0.023*"queen" 
0.023*"come"')

'0.052*"me" 
0.021*"want" 
0.019*"no"  
0.019*"dont"
0.013*"see"')